syncstorage-mysql/src/error.rs

```diff
@@ -0,0 +1,138 @@
+use std::fmt;
+
+use backtrace::Backtrace;
+use http::StatusCode;
+use syncserver_common::{from_error, impl_fmt_display, ReportableError};
+use syncserver_db_common::error::MysqlError;
+use syncstorage_db_common::error::{DbErrorIntrospect, SyncstorageDbError};
+use thiserror::Error;
+
+/// An error type that represents any MySQL-related errors that may occur while processing a
+/// syncstorage request. These errors may be application-specific or lower-level errors that arise
+/// from the database backend.
+#[derive(Debug)]
+pub struct DbError {
+    kind: DbErrorKind,
+    pub status: StatusCode,
+    pub backtrace: Backtrace,
+}
+
+impl DbError {
+    pub fn batch_not_found() -> Self {
+        DbErrorKind::Common(SyncstorageDbError::batch_not_found()).into()
+    }
+
+    pub fn bso_not_found() -> Self {
+        DbErrorKind::Common(SyncstorageDbError::bso_not_found()).into()
+    }
+
+    pub fn collection_not_found() -> Self {
+        DbErrorKind::Common(SyncstorageDbError::collection_not_found()).into()
+    }
+
+    pub fn conflict() -> Self {
+        DbErrorKind::Common(SyncstorageDbError::conflict()).into()
+    }
+
+    pub fn internal(msg: String) -> Self {
+        DbErrorKind::Common(SyncstorageDbError::internal(msg)).into()
+    }
+
+    pub fn quota() -> Self {
+        DbErrorKind::Common(SyncstorageDbError::quota()).into()
+    }
+}
+
+#[derive(Debug, Error)]
+enum DbErrorKind {
+    #[error("{}", _0)]
+    Common(SyncstorageDbError),
```

Comment:
I struggled a bit with the naming here. `Common` is supposed to be a Syncstorage database error that is not specific to the database backend